### PR TITLE
Fix retrieving the sessions using the user id in user profile

### PR DESCRIPTION
**Pull request by ThaminduDilshan**
### Proposed changes in this pull request
For the federated users, session-related information is persisted under a different user id. So, when we try to retrieve the sessions (GET /users/v1/{user_id}/sessions) [1] using the user id in the user profile it won't return any results.

In the IDN_AUTH_USER and IDN_AUTH_USER_SESSION_MAPPING tables, it's maintaining a different user id, not the one in user profile.

**Changes**
- Check for federated associations for the given userId and retrieve session using the retrieved id.
- Add idpName to the sessions API responses.

### Related issue
- https://github.com/wso2/product-is/issues/12169

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/association/federation/FederatedAssociationManager.java [hunk 1]
```diff
@@ -88,6 +88,17 @@ String getUserForFederatedAssociation(String tenantDomain, String idpName, Strin
...
 
+    /**
+     * Return an array of federated associations associated with the given user.
+     *
+     * @param tenantId Id of the tenant.
+     * @param domainFreeUsername username without domain.
+     * @return An array of federated associations which contains the federated identifier info.
+     * @throws FederatedAssociationManagerException
+     */
+    FederatedAssociation[] getFederatedAssociationsOfUser(int tenantId, String domainFreeUsername) throws
```
[the remark this answers is not in the document]
Comment (ThaminduDilshan) [reply components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/association/federation/FederatedAssociationManager.java]: Addressed with 4d8a43e78446f7af86d07d8b4e9eb3b63509d8e2

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java [hunk 2]
```diff
@@ -4367,4 +4368,60 @@ public boolean isClaimReferredByAnyIdp(Connection dbConnection, String claimUri,
             }
         }
     }
+
```
[the remark this answers is not in the document]
Comment (ThaminduDilshan) [reply components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java:4371]: Addressed with 4d8a43e78446f7af86d07d8b4e9eb3b63509d8e2

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/CacheBackedIdPMgtDAO.java [hunk 3]
```diff
@@ -937,4 +938,10 @@ private String getIDPIssuerName(IdentityProvider identityProvider) {
         }
         return null;
     }
+
+    public List<IdentityProvider> getIdPsById(Connection dbConnection, int tenantId, Set<String> idpIds)
```
[the remark this answers is not in the document]
Comment (ThaminduDilshan) [reply components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/CacheBackedIdPMgtDAO.java:951]: Addressed with 4d8a43e78446f7af86d07d8b4e9eb3b63509d8e2

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java [hunk 4]
```diff
@@ -4367,4 +4368,69 @@ public boolean isClaimReferredByAnyIdp(Connection dbConnection, String claimUri,
...
+    /**
+     * Method that retrieves identityProvider details of a idpId list.
+     *
+     * @param dbConnection Optional DB connection.
+     * @param tenantId Tenant id.
+     * @param idpIds Set of identity provider ids.
+     * @return A list of identity providers.
+     * @throws IdentityProviderManagementException
+     */
+    public List<IdentityProvider> getIdPsById(Connection dbConnection, int tenantId, Set<String> idpIds)
```
Comment (ThaminduDilshan) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java:4381]: 1. Let's modify this method to get only IdpNames.
2. Let's remove connection parameter

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java [hunk 5]
```diff
@@ -4367,4 +4368,69 @@ public boolean isClaimReferredByAnyIdp(Connection dbConnection, String claimUri,
...
+            dbConnInitialized = true;
+        }
+
+        try {
+            String placeholder = String.join(", ", idpIds);
+            String sqlStmt = IdPManagementConstants.SQLQueries.GET_IDPS_BY_IDP_ID_LIST.replace(
+                    SCOPE_LIST_PLACEHOLDER, placeholder);
+            prepStmt = dbConnection.prepareStatement(sqlStmt);
+            prepStmt.setInt(1, tenantId);
+            prepStmt.setInt(2, MultitenantConstants.SUPER_TENANT_ID);
```
Comment (ThaminduDilshan) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java:4400]: We don't need to filter by super tenant here

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java [hunk 6]
```diff
@@ -122,6 +135,43 @@ private String resolveUserIdFromUsername(int tenantId, String userStoreDomain, S
...
 
+    /**
+     * Retrieves the username of the given userId.
+     *
+     * @param tenantId Id of the tenant domain of the user.
+     * @param userId Id of the user.
+     * @return username.
+     * @throws UserSessionException
+     */
+    private String getUsernameFromUserId(int tenantId, String userId) throws
```
Comment (ThaminduDilshan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java:146]:
```suggestion
    private String getUsernameFromUserId(String userId, int tenantId) throws
```

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java [hunk 7]
```diff
@@ -122,6 +135,43 @@ private String resolveUserIdFromUsername(int tenantId, String userStoreDomain, S
...
+            try {
+                if (userStoreManager instanceof AbstractUserStoreManager) {
+                    return ((AbstractUserStoreManager) userStoreManager).getUserNameFromUserID(userId);
+                }
+                if (log.isDebugEnabled()) {
+                    log.debug("Provided user store manager for the tenantId: " + tenantId + ", is not an instance " +
+                            "of the AbstractUserStore manager.");
+                }
+                throw new UserSessionException("Unable to get the username for the userId: " + userId + ".");
+            } catch (org.wso2.carbon.user.core.UserStoreException e) {
```
Comment (ThaminduDilshan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java:161]: Check and verify two different exceptions with `UserStoreException`. Remove the redundent

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java [hunk 8]
```diff
@@ -486,4 +687,141 @@ private SessionManagementClientException handleSessionManagementClientException(
...
+
+    /**
+     * This method checks whether federated associations exist for the given userId and if so returns the
+     * internal userId stored in IDN_AUTH_USER table.
+     *
+     * @param userId User Id.
+     * @param tenantId Tenant Id.
+     * @return User id to which sessions are stored against.
+     */
+    private String getUserIdFromFederatedMapping(int tenantId, String userId) throws UserSessionException {
```
Comment (ThaminduDilshan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java:699]:
```suggestion
    private String getAuthSessionUserIdFromAssociatedMapping(int tenantId, String userId) throws UserSessionException {
```
Think on a better name

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java [hunk 9]
```diff
@@ -486,4 +687,141 @@ private SessionManagementClientException handleSessionManagementClientException(
...
+    }
+
+    /**
+     * Retrieve and set identity provider information to the session object. Currently, only the idpName is set.
+     *
+     * @param tenantDomain Tenant domain of the user.
+     * @param userSession User session containing idpId.
+     * @throws UserSessionException Exception is thrown if any error occurred.
+     */
+    private void parseIdpInfoToSessionResponse(String tenantDomain, UserSession userSession)
```
Comment (ThaminduDilshan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java:794]: Reuse previous method by creating a list with singleton class

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SQLQueries.java [hunk 10]
```diff
@@ -327,4 +327,8 @@ public class SQLQueries {
             "WHERE EXISTS (SELECT SESSION_ID, TIME_CREATED FROM SESSIONS s WHERE ss.SESSION_ID = s.SESSION_ID AND " +
             "ss.TIME_CREATED = s.TIME_CREATED) AND OPERATION = ''STORE'' " +
             "AND SESSION_TYPE = ''AppAuthFrameworkSessionContextCache''{3} ORDER BY TIME_CREATED {4}";
+
+    public static final String SQL_GET_IDP_ID_BY_USER_ID = "SELECT IDP_ID FROM IDN_AUTH_USER WHERE USER_ID = ?";
+    public static final String SQL_GET_IDP_IDS_BY_USER_ID_LIST = "SELECT USER_ID, IDP_ID FROM IDN_AUTH_USER WHERE " +
```
Comment (ThaminduDilshan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SQLQueries.java:332]: These two queries can be removed by returning idpIds by `getUserIdFromFederatedMapping`

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/impl/UserSessionDAOImpl.java [hunk 11]
```diff
@@ -109,7 +113,7 @@ public UserSession getSession(String sessionId) throws SessionManagementServerEx
     }
 
     @Override
-    public Optional<UserSession> getSession(String userId, String sessionId) throws SessionManagementServerException {
+    public UserSession getSession(String userId, String sessionId) throws SessionManagementServerException {
```
[the remark this answers is not in the document]
Comment (ThaminduDilshan) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/impl/UserSessionDAOImpl.java:116]: Let's remove this change and reuse Optional from Service layer

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java (hunk 7 above)
Comment (ThaminduDilshan) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java:161]: These are existing methods and are returning these two types of `UserStoreException`

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java [hunk 12]
```diff
@@ -122,6 +135,43 @@ private String resolveUserIdFromUsername(int tenantId, String userStoreDomain, S
...
+     */
+    private String getUsernameFromUserId(int tenantId, String userId) throws
+            UserSessionException {
+
+        try {
+            UserStoreManager userStoreManager = FrameworkServiceComponent.getRealmService()
+                    .getTenantUserRealm(tenantId).getUserStoreManager();
+            try {
+                if (userStoreManager instanceof AbstractUserStoreManager) {
+                    return ((AbstractUserStoreManager) userStoreManager).getUserNameFromUserID(userId);
```
[the remark this answers is not in the document]
Comment (ThaminduDilshan) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java:154]: Even though there's a existing `getFederatedAssociationsOfUser` method in `FederatedAssociationManager`, it internally query for all the `IdentityProvider` details. Hence keeping the newly added method in `FederatedAssociationManagerImpl` by modifying it to only return the required details.
[ref1](https://github.com/wso2/carbon-identity-framework/blob/master/components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/association/federation/FederatedAssociationManagerImpl.java#L148), [ref2](https://github.com/wso2/carbon-identity-framework/blob/master/components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/association/federation/FederatedAssociationManagerImpl.java#L456), [ref3](https://github.com/wso2/carbon-identity-framework/blob/master/components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java#L1994)

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java [hunk 13]
```diff
@@ -179,16 +229,68 @@ public List<UserSession> getSessionsByUserId(String userId) throws SessionManage
...
+        // First check whether a federated association exists for the userId.
+        try {
+            int tenantId = getTenantId(tenantDomain);
+            String fedAssociatedUserId = getUserIdFromFederatedMapping(tenantId, userId);
+
+            if (StringUtils.isNotEmpty(fedAssociatedUserId)) {
+                userSessions = getActiveSessionList(
+                        fedAssociatedUserId, getSessionIdListByUserId(fedAssociatedUserId));
+            } else {
+                userSessions = getActiveSessionList(userId, getSessionIdListByUserId(userId));
```
Comment (ThaminduDilshan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java:244]: Requires to merge these two results to accommodate linking local users to another user.
Ex: JIT provisioned user entering a password and allowing it to login with local userId.

File: components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/association/federation/FederatedAssociationManager.java [hunk 14]
```diff
@@ -88,6 +88,20 @@ String getUserForFederatedAssociation(String tenantDomain, String idpName, Strin
...
 
+    /**
+     * Return an array of federated associations associated with the given user.
+     *
+     * @param tenantId Id of the tenant.
+     * @param domainFreeUsername username without domain.
+     * @return An array of federated associations which contains the federated identifier info.
+     * @throws FederatedAssociationManagerException
+     */
+    default FederatedAssociation[] getFederatedAssociationsOfUser(int tenantId, String domainFreeUsername) throws
```
[the remark this answers is not in the document]
Comment (ThaminduDilshan) [reply components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/association/federation/FederatedAssociationManager.java]: Even though there's a existing `getFederatedAssociationsOfUser` method in `FederatedAssociationManager`, it internally query for all the `IdentityProvider` details. Hence keeping the newly added method in `FederatedAssociationManagerImpl` by modifying it to only return the required details.
[ref1](https://github.com/wso2/carbon-identity-framework/blob/master/components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/association/federation/FederatedAssociationManagerImpl.java#L148), [ref2](https://github.com/wso2/carbon-identity-framework/blob/master/components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/association/federation/FederatedAssociationManagerImpl.java#L456), [ref3](https://github.com/wso2/carbon-identity-framework/blob/master/components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java#L1994)

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java (hunk 13 above)
Comment (ThaminduDilshan) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java]: Addressed with 7056d26ef22a1c38205b79e2a0143791c98761b8

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/UserSessionDAO.java [hunk 15]
```diff
@@ -61,7 +61,7 @@ default UserSession getSession(String sessionId) throws SessionManagementServerE
      */
     default Optional<UserSession> getSession(String userId, String sessionId) throws SessionManagementServerException {
 
-        return Optional.empty();
+        return null;
```
Comment (senthalan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/UserSessionDAO.java:64]: `return Optional.empty();`

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/impl/UserSessionDAOImpl.java [hunk 16]
```diff
@@ -154,6 +156,7 @@ public Optional<UserSession> getSession(String userId, String sessionId) throws
 
             if (!applicationList.isEmpty()) {
                 userSession.setApplications(applicationList);
+
```
Comment (senthalan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/impl/UserSessionDAOImpl.java:159]: unwanted new line

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/impl/UserSessionDAOImpl.java [hunk 17]
```diff
@@ -98,6 +98,7 @@ public UserSession getSession(String sessionId) throws SessionManagementServerEx
...
```
Comment (senthalan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/impl/UserSessionDAOImpl.java:101]: unwanted new line

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/UserSessionDAO.java (hunk 15 above)
Comment (senthalan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/UserSessionDAO.java:64]: `return Optional.empty();`

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/UserSessionDAO.java [hunk 18]
```diff
@@ -59,9 +58,9 @@ default UserSession getSession(String sessionId) throws SessionManagementServerE
      * @return User session.
      * @throws SessionManagementServerException
      */
-    default Optional<UserSession> getSession(String userId, String sessionId) throws SessionManagementServerException {
+    default UserSession getSession(String userId, String sessionId) throws SessionManagementServerException {
```
[the remark this answers is not in the document]
Comment (ThaminduDilshan) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/UserSessionDAO.java]: fixed

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/UserSessionDAO.java (hunk 15 above)
Comment (ThaminduDilshan) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/UserSessionDAO.java]: Addressed with 44011d5a83074e785d38b41416cf8ea6255b2113

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/impl/UserSessionDAOImpl.java (hunk 16 above)
Comment (ThaminduDilshan) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/impl/UserSessionDAOImpl.java]: Addressed with 44011d5a83074e785d38b41416cf8ea6255b2113

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/impl/UserSessionDAOImpl.java (hunk 17 above)
Comment (ThaminduDilshan) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/impl/UserSessionDAOImpl.java]: Addressed with 44011d5a83074e785d38b41416cf8ea6255b2113

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/UserSessionDAO.java (hunk 15 above)
Comment (ThaminduDilshan) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/UserSessionDAO.java]: Addressed with 44011d5a83074e785d38b41416cf8ea6255b2113

**Review by senthalan — APPROVED** [review]

**Comment by ThaminduDilshan** [comment]
**PR builder test failure**

```
org.wso2.identity.integration.test.rest.api.user.approval.v1.UserMeApprovalTest.testListTasksWhenAvailable(org.wso2.identity.integration.test.rest.api.user.approval.v1.UserMeApprovalTest)
[ERROR] Run 1: UserMeApprovalTest.testListTasksWhenAvailable:141 1 expectation failed.
JSON path findAll{ it.presentationName == 'TestWorkflowAddUserForRestTask' }.size() doesn't match.
Expected: is <3>
Actual: <1>
 
[INFO] Run 2: PASS
[INFO] 
[INFO] 
[ERROR] Tests run: 1891, Failures: 1, Errors: 0, Skipped: 6
```
Shouldn't be related

**Pull request merged**